backend/src/score.ts

```diff
@@ -370,29 +480,39 @@ const scoreCountry = (countryA: string|string[]|RequestField, countryB: string|s
     if (typeof(countryA) === 'string') {
         const countryNormA = countryNorm(countryA) as string;
         if (typeof(countryB) === 'string') {
-            return fuzzyScore(countryNormA, countryNorm(countryB) as string);
+            return fuzzyRatio(countryNormA, countryNorm(countryB) as string, fuzzSetRatio);
         } else {
-            return Math.max(...countryB.map(country => fuzzyScore(countryNormA, countryNorm(country) as string)));
+            return Math.max(...countryB.map(country => fuzzyRatio(countryNormA, countryNorm(country) as string, fuzzSetRatio)),
+                fuzzSetRatio(countryNormA, countryB.join(' '))
+                );
         }
     } else {
         const countryNormB = countryNorm(countryB);
-        return Math.max(...(countryA as string[]).map(country => scoreCountry(countryNorm(country), countryNormB)));
+        return Math.max(...(countryA as string[]).map(country => scoreCountry(countryNorm(country), countryNormB)),
+            fuzzSetRatio((countryA as string[]).join(' '), Array.isArray(countryNormB) ? countryNormB.join(' ') : countryNormB)
+        );
     }
 }
 
 
 const scoreLocation = (locA: Location, locB: Location): any => {
     const score: any = {};
-    if (locB.country && (scoreCountry('FRANCE', locB.country as string|string[]) === 1)) {
+    const BisFrench = locB.country && (locB.countryCode === 'FRA');
+    if (BisFrench) {
         if (normalize(locA.country as string|string[])) {
             score.country = scoreCountry(locA.country, tokenize(locB.country as string) as string|string[]);
         }
         if (normalize(locA.city as string|string[]) && locB.city) {
             score.city = scoreCity(locA.city, locB.city as string|string[]);
         }
         if (normalize(locA.departmentCode as string|string[]) && locB.departmentCode) {
-            if (locB.country && (scoreCountry('FRANCE', locB.country as string|string[]) === 1)) {
-                score.department = (locA.departmentCode === locB.departmentCode) ? 1 : minDepScore;
+            if (BisFrench) {
+                if (locA.departmentCode !== '99') {
+                score.department = (locA.departmentCode === locB.departmentCode) ? 1 :
+                    ( ( (score.city === 1) && (locB.departmentCode === '75') && (['78','91','92','93','94','95'].indexOf(locA.departmentCode as string)) ) ? 1 : minDepScore);
+                } else {
+                    score.country = minLocationScore;
+                }
             }
```

Comment:
I think this condition is not necessary, we are already inside this line

```suggestion
                if (locA.departmentCode !== '99') {
                score.department = (locA.departmentCode === locB.departmentCode) ? 1 :
                    ( ( (score.city === 1) && (locB.departmentCode === '75') && (['78','91','92','93','94','95'].indexOf(locA.departmentCode as string)) ) ? 1 : minDepScore);
                } else {
                    score.country = minLocationScore;
                }
```

Reply:
Indeed some test weren't well ordered, i did change the whole block